atd-vzd/migrations/default/1697144772669_add_ems_trigger_function/up.sql

```diff
@@ -0,0 +1,26 @@
+CREATE OR REPLACE FUNCTION public.ems_incidents_trigger()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+  update ems__incidents set
+    austin_full_purpose = (
+      select ST_Contains(jurisdiction.geometry, incidents.geometry)

```

Comment:
is there a concern that geometry would ever be null/missing?

Reply:
I checked this table in production and there are a handful of nulls for geometry. Ends up with a nulled `austin_full_purpose` column. That should be an easy fix.

Reply:
On second thought, I noticed that this is comes from using `ST_SetSRID` and `ST_Point` in the ETL. Maybe we should have some upstream changes to null these out there. Calling @frankhereford for phone a friend ☎️

Reply:
I think this function as written will work fine, because if `ST_Contains()` gets a `null` argument the result is null. Here's a test fixture:

```
SELECT ST_Contains(
    ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))'), 
    ST_GeomFromText('POINT(5 5)')
) AS result;

SELECT ST_Contains(
    ST_GeomFromText('POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))'), 
    null
) AS result;

SELECT ST_Contains(
    null, 
    ST_GeomFromText('POINT(5 5)')
) AS result;
```

To Mike's point about how these bogus `0,0` points have snuck into the DB - I do think we should fix that. `0,0` is a valid point, which is why it's slipping through the GIS functions..., but we should trap it as invalid because it's just not one we're going to run into here in Austin when using SRID 4326, 3857, or 2277. I think that a new issue though, not one we should try to pick off in this work.